server/app/auth/CiviFormProfileMerger.java

```diff
@@ -81,10 +81,22 @@ private Optional<CiviFormProfile> mergeApplicantAndGuestProfile(
         // Merge the two applicants and prefer the newer one.
         guestProfile = Optional.of(mergeProfiles(applicantInDatabase.get(), guestProfile.get()));
       }
+      // Ideally, the applicant id would already be populated in `guestProfile`. However, there
+      // could be profiles in user sessions that were created before we started populating this
+      // info.
+      storeApplicantIdInProfile(guestProfile.orElseThrow(), applicantInDatabase.orElseThrow());
```

Comment:
question (blocking): when might this throw?

Reply:
These two accesses are guaranteed not to throw by the surrounding logic.

Reply:
Thanks!